Add the `phred_quality_score` method to the Python `Record` class

apcamargo opened the pull request:
This PR introduces the `decode_phred` function that takes in a string representing Phred-encoded quality scores and returns a tuple of integers representing the quality scores themselves. The function uses Phred+33 by default and allows the user to use Phred+64 via the `base_64` parameter.

```py
>>> from needletail import decode_phred
>>> decode_phred("#</</BBFFFBF<")
(2, 27, 14, 27, 14, 33, 33, 37, 37, 37, 33, 37, 27)
```

audy reviewed (changes requested):
@apcamargo I think it's probably worth adding this to `SequenceRecord` in the underlying Rust implementation.

apcamargo commented:
Thanks, @audy. I don't have much familiarity with the Rust library. Could you provide any pointers of where this should go? Maybe a method for `SequenceRecord` in [`record.rs`](https://github.com/onecodex/needletail/blob/master/src/parser/record.rs)?

audy commented:
> Thanks, @audy. I don't have much familiarity with the Rust library. Could you provide any pointers of where this should go? Maybe a method for `SequenceRecord` in [`record.rs`](https://github.com/onecodex/needletail/blob/master/src/parser/record.rs)?

I think the best place is probably `parser/record.rs`. `parser/fastq.rs` has methods for returning data straight from the file buffer. But since this requires some transformation of that data, I think `parser/record.rs` makes the most sense.

apcamargo commented:
@audy I updated this PR to use the functions from the Rust library. I also moved the function out of the `Record` class to keep it consistent with `reverse_complement`. Tests were updated accordingly.

`normalize` is still the odd one out since it also exists as a method that modifies the object. Maybe we can think about making things more consistent in a future breaking update?

audy reviewed (approved):
Looks good to me!

audy commented:
> `normalize` is still the odd one out since it also exists as a method that modifies the object. Maybe we can think about making things more consistent in a future breaking update?

I think I would prefer a method that returns a new `SequenceRecord` with a normalized sequence (same in the underlying Rust implementation) rather than modifying the record in place. What do you think?

apcamargo commented:
Agreed! Let me know how you prefer to do this, since it would break the API
